pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -349,6 +364,11 @@ func (c *VSphereController) blockUpgradeOrDegradeCluster(
 		// Set Upgradeable: true with an extra message
 		updateError := c.updateConditions(ctx, c.name, result, status, operatorapi.ConditionFalse)
 		return updateError, true, true
+	case checks.ClusterCheckUpgradesBlockedViaAdminAck:
+		clusterCondition = "upgrade_blocked_admin_ack"
+		utils.InstallErrorMetric.WithLabelValues(string(result.CheckStatus), clusterCondition).Set(1)
+		updateError := c.addRequiresAdminAck(ctx, result, c.name)
+		return updateError, false, false
```

Comment:
Where is the admin-ack _removed_ when user moves all PVs to CSI?

Reply:
Added some code for it for removing the admin-ack keys when conditions are right. But this needs to be carefully reviewed. See - https://github.com/openshift/vmware-vsphere-csi-driver-operator/pull/169/files#diff-bd7b198fe282f7e27754079508639882be175cc903f563fb69f2c89fc6875a2aR319